frontend/src/components/common/form/ParentSelect.tsx

```diff
@@ -0,0 +1,132 @@
+import { getIn, useFormikContext } from 'formik';
+import { groupBy, sortBy } from 'lodash';
+import React, { Fragment, useEffect, useState } from 'react';
+import { Highlighter, Menu, MenuItem } from 'react-bootstrap-typeahead';
+import { Label } from '../Label';
+import { SelectOption, SelectOptions } from './Select';
+import { TypeaheadField } from './Typeahead';
+
+/** Component used to group children items with their parent.
+ *  Can click the parent item as a value or the subsequent children
+ **/
+
+interface IParentSelect {
+  /** specify the field that is being accessed */
+  field: string;
+  /** give the component the desired options to group by parent with */
+  options: SelectOptions;
+  /** if you would like placeholder text within the input */
+  placeholder?: string;
+  /** filterBy is optional - if used can determine what you can filter with */
+  filterBy?: string[];
+  /** provide the opitional label text*/
+  label?: string;
+  /** flag used to control whehter to display red astrix indicating required */
+  required?: boolean;
```

Comment:
Since you are in dev you should be aware that we aren't using the red asterix to denote required inputs anymore, we are now highlighting the field in light blue. see the property detail form in pims-dev as an example.